README.md

```diff
@@ -274,3 +274,63 @@ failed:
 ```
 
 </details>
+
+## Logic preview
+<details>
+<summary>
+Logic diagrams
+</summary>
+
+```mermaid
+graph TD
+    A[WORKFLOW DOKKU] -->|Deploy to Dokku| B(activate tarantool instance 'job rerunner')
+    B --> C[Runs function 'webhook_handler', a function that decides <br />whether to restart jobs or not]
+    C -->|perform check 1| D{needs_restart?}
+    C -->|perform check 2| E{is_job_failed?}
+    D --> |True| F
+    E --> |True| F[run function re_run_failed_jobs <br/> and return status 200]
+    D --> |False| G
+    E --> |False| G[return status 200]
+```
+
+needs_restart()
+```mermaid
+graph TD
+    A[needs_restart function] -->|accept job_id| B{check tarantool space 'jobs' is job exist?}
+    B --> |true| C{check it's count < 3}
+    B --> |false| D[insert into space <br/> new data about job and return False]
+    C --> E[increment it's count and return True]
+```
+
+is_job_failed()
+```mermaid
+graph TD
+    A[is_job_failed function] -->|accept some data - lua-table| B{check incoming data type,<br/> is incoming data type is 'table'}
+    B --> |true| C{check if data.workflow_job is 'table' type}
+    B --> |false| D[return False]
+    C --> |true|E[return boolean result <br/> of checking workflow_job.conclusion == 'failure']
+    C --> |false|F[return False]
+```
+
+re_run_failed_jobs()
+```mermaid
+graph TD
+    A[re_run_failed_jobs function] -->|accept repo address and run_id| B[send post-request to rerun 'run' and set attempts = 1]
+    B --> B[while attempts < 12]
+    B --> |true| C{if request status equal to 403}
+    B --> |false| End
+    C --> |true| D[fiber.sleep for 60 seconds <br/> then send request to rerun and 1 to attempts]
+    D --> End
+```
+
+</details>
+
+## Status codes
+Job rerunner returns some status code while working on payload from GitHub.
+
+| Status code | Meaning                                                                          |
+|-------------|----------------------------------------------------------------------------------|
+| 204         | Got empty payload or overall run attempts more or equal to 4                     |
+| 200         | Job action is 'queued' or 'in_progress', Rerunner server is Active               |
+| 201         | Job action is 'completed' and information about run deleted from Tarantool space |
+| 404         | No records posted to rerunner                                                    |
```

Comment:
```suggestion
| 204         | Got empty payload or overall run attempts more or equal to 4. No action.                     |
| 200         | Job action is 'queued' or 'in_progress'. Recording jobs in this workflow run.               |
| 201         | Job action is 'completed'. Restarting the workfow. |
| 404         | Job action is 'completed', but there's no previous record about it.                                                    |
```

Reply:
Overall, I propose using codes in this way:

* 204 No Content — no payload or no action required.
* 202 Accepted — recorded the `queued` or `completed` jobs and waiting for results.
* 201 Created — final job completed, making a restart.
* 200 OK — final job completed, but no action is required.
* 404 — cannot find the record for this workflow. (This should not happen).
https://en.wikipedia.org/wiki/List_of_HTTP_status_codes#2xx_success

Reply:
> Overall, I propose using codes in this way:
> 
>     * 204 No Content — no payload or no action required.
> 
>     * 202 Accepted — recorded the `queued` or `completed` jobs and waiting for results.
> 
>     * 201 Created — final job completed, making a restart.
> 
>     * 200 OK — final job completed, but no action is required.
> 
>     * 404 — cannot find the record for this workflow. (This should not happen).
>       [en.wikipedia.org/wiki/List_of_HTTP_status_codes#2xx_success](https://en.wikipedia.org/wiki/List_of_HTTP_status_codes#2xx_success)

Reply:
Done